handlers/mparticle-api/src/index.ts

```diff
@@ -96,7 +90,8 @@ const router = new Router([
         httpMethod: 'POST',
         path: '/data-subject-requests/{requestId}/callback',
         handler: routerHandler(async (event, parsed) => {
-            const callbackValidationResult = await validateDataSubjectRequestCallback(event.headers['x-opendsr-processor-domain'], event.headers['x-opendsr-signature'], event.body);
+            const getHeader = (key: string) => Object.entries(event.headers).find(([k]) => k.toLowerCase() === key.toLowerCase())?.[1];
+            const callbackValidationResult = await validateDataSubjectRequestCallback(getHeader('x-opendsr-processor-domain'), getHeader('x-opendsr-signature'), event.body);
```

Comment:
nit pick/optional:
I guess this change is because header names are not case sensitive - It is a bit harder to follow the `k` vs `key` and the `?.[1]`, I think it'll be slightly clearer to preprocess them and keep the later logic the same, what do you think?
```suggestion
            const headers = mapValues(event.headers, v => v.toLowerCase());
            const callbackValidationResult = await validateDataSubjectRequestCallback(headers['x-opendsr-processor-domain'], headers['x-opendsr-signature'], event.body);
```
and you can import mapValues from arrayFunctions.ts